etna/models/base.py

```diff
@@ -450,4 +467,314 @@ def get_model(self) -> Any:
+
+        Parameters
+        ----------
+        ts:
+            TSDataset with features
+
+        Returns
+        -------
+        :
+            Model after fit
+        """
+        ts_torch = ts.to_torch_dataset(self.make_samples, dropna=True)
+        self.raw_fit(ts_torch)
+        return self
+
+    def raw_fit(self, torch_dataset: "Dataset") -> "DeepBaseModel":
+        """Fit model on torch like Dataset.
+
+        Parameters
+        ----------
+        torch_dataset: Dataset
+            Torch like dataset for model fit
+
+        Returns
+        -------
+        :
+            Model after fit
+        """
+        if self.split_params:
+            if isinstance(torch_dataset, Sized):
+                tsdataset_length = len(torch_dataset)
+            else:
+                tsdataset_length = self.split_params["dataset_size"]
+            train_frac = self.split_params["train_frac"]
+            train_dataset, val_dataset = random_split(
+                torch_dataset,
+                lengths=[int(train_frac * tsdataset_length), tsdataset_length - int(train_frac * tsdataset_length)],
+                generator=self.split_params.get("generator"),
+            )
+            train_dataloader = DataLoader(
+                train_dataset, batch_size=self.train_batch_size, shuffle=True, **self.train_dataloader_params
+            )
+            val_dataloader: Optional[DataLoader] = DataLoader(
+                val_dataset, batch_size=self.test_batch_size, shuffle=False, **self.val_dataloader_params
+            )
+        else:
+            train_dataloader = DataLoader(
+                torch_dataset, batch_size=self.train_batch_size, shuffle=True, **self.train_dataloader_params
+            )
+            val_dataloader = None
+
+        if "logger" not in self.trainer_params:
+            self.trainer_params["logger"] = tslogger.pl_loggers
+        else:
+            self.trainer_params["logger"] += tslogger.pl_loggers
+
+        trainer = Trainer(**self.trainer_params)
+        trainer.fit(self, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader)
+        return self
+
+    def raw_predict(self, torch_dataset: "Dataset") -> Dict[Tuple[str, str], np.ndarray]:
+        """Make inference on torch like Dataset.
+
+        Parameters
+        ----------
+        torch_dataset: Dataset
+            Torch like dataset for model inference
+
+        Returns
+        -------
+        :
+            Dictionary with predictions
+        """
+        test_dataloader = DataLoader(
+            torch_dataset, batch_size=self.test_batch_size, shuffle=False, **self.test_dataloader_params
+        )
+
+        predictions_dict = dict()
+        self.eval()
+        with torch.no_grad():
+            for batch in test_dataloader:
+                segments = batch["segment"]
+                predictions = self(batch)
+                predictions_array = predictions.numpy()
+                for idx, segment in enumerate(segments):
```

Comment:
May it is better to store the predictions somewhere and create the dictionary in the end of the method like this `dict(zip(segments, predictions))`

Reply:
I guess it makes code more implicit.
and we can't use zip straightforward cause we will have list of ndarrays at the end

Reply:
If it does not significantly decrease the performance let's leave this implementation